driver-reactive-streams/src/test/functional/com/mongodb/reactivestreams/client/syncadapter/SyncMongoCursor.java

```diff
@@ -27,8 +28,11 @@
 class SyncMongoCursor<T> implements MongoCursor<T> {
     private final Iterator<T> iterator;
 
-    SyncMongoCursor(final Publisher<T> publisher) {
-        iterator = Flux.from(publisher).toIterable().iterator();
+    SyncMongoCursor(final Publisher<T> publisher, final @Nullable Integer batchSize) {
+        iterator = (batchSize == null
```

Comment:
Does this make any difference?

Reply:
Without this code we lose the explicitly set batch size and the Reactor uses (requests) 256 instead as its default.

Reply:
Ack